nodejs/aws-infra/experimental/clusterService.ts

```diff
@@ -0,0 +1,324 @@
+        assignPublicIp?: pulumi.Input<boolean>;
+        securityGroups?: pulumi.Input<pulumi.Input<string>[]>;
+        subnets: pulumi.Input<pulumi.Input<string>[]>;
+    }>;
+    /**
+     * Service level strategy rules that are taken into consideration during task placement. List
+     * from top to bottom in order of precedence. The maximum number of `ordered_placement_strategy`
+     * blocks is `5`. Defined below.
+     */
+    orderedPlacementStrategies?: pulumi.Input<pulumi.Input<{
+        field?: pulumi.Input<string>;
+        type: pulumi.Input<string>;
+    }>[]>;
+    /**
+     * rules that are taken into consideration during task placement. Maximum number of
+     * `placement_constraints` is `10`. Defined below.
+     */
+    placementConstraints?: pulumi.Input<pulumi.Input<{
+        expression?: pulumi.Input<string>;
+        type: pulumi.Input<string>;
+    }>[]>;
+    /**
+     * **Deprecated**, use `ordered_placement_strategy` instead.
+     */
+    placementStrategies?: pulumi.Input<pulumi.Input<{
+        field?: pulumi.Input<string>;
+        type: pulumi.Input<string>;
+    }>[]>;
+    /**
+     * The scheduling strategy to use for the service. The valid values are `REPLICA` and `DAEMON`.
+     * Defaults to `REPLICA`. Note that [*Fargate tasks do not support the `DAEMON` scheduling
+     * strategy*](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/scheduling_tasks.html).
+     */
+    schedulingStrategy?: pulumi.Input<string>;
+    /**
+     * The service discovery registries for the service. The maximum number of `service_registries` blocks is `1`.
+     */
+    serviceRegistries?: pulumi.Input<{
+        containerName?: pulumi.Input<string>;
+        containerPort?: pulumi.Input<number>;
+        port?: pulumi.Input<number>;
+        registryArn: pulumi.Input<string>;
+    }>;
+
+    // Changes we made to the core args type.
+
+    /**
+     * Cluster this service will run in.
+     */
+    cluster: mod.Cluster;
+
+    /**
+     * The task definition to create the service from.
+     */
+    taskDefinition: mod.TaskDefinition;
+
+    /**
+     * The number of instances of the task definition to place and keep running. Defaults to 1. Do
+     * not specify if using the `DAEMON` scheduling strategy.
+     */
+    desiredCount?: pulumi.Input<number>;
+
+    /**
+     * The launch type on which to run your service. The valid values are `EC2` and `FARGATE`.
+     * Defaults to `EC2`.
+     */
+    launchType?: pulumi.Input<"EC2" | "FARGATE">;
+
+    os?: pulumi.Input<"linux" | "windows">;
+
+    /**
+     * Wait for the service to reach a steady state (like [`aws ecs wait
+     * services-stable`](https://docs.aws.amazon.com/cli/latest/reference/ecs/wait/services-stable.html))
+     * before continuing. Defaults to `true`.
+     */
+    waitForSteadyState?: pulumi.Input<boolean>;
+
+    /**
+     * Optional auto-scaling group for the cluster.
+     */
+    autoScalingGroup?: mod.ClusterAutoScalingGroup;
```

Comment:
the autoscaling group is really only needed so that teh Service can take a 'dependson' relationship purely so it can wait for this to be ready.  it seems unfortunate.  One way we could get around this would be to have a ClusterAutoScalingGroup register itself with the Cluster.  Then, the Service could just take a dependsOn on that implicitly without people needing to know to pass this in.  Thoughts @lukehoban ?